howto/how-to-set-up-a-blaster.md

```diff
@@ -0,0 +1,331 @@
+
+## Required Defines
+
+### NUM_BLADES
+Similarly to the saber version, each "blade" is a string of pixels (or a single one) that will have a styled defined for each soundfont. Remember that any accent, crystal chamber and subblade you define counts towards a different blade.
+
+```cpp
+#define NUM_BLADES 2
+```
+
+### NUM_BUTTONS
+And we must also specify how many buttons we have. Remember that for `blaster.h` prop, your first button is Fire, the second is Mode button and then you can add Power, Reload or Clip.
+
+```cpp
+#define NUM_BUTTONS 2
+```
+
+### CLASH_THRESHOLD_G
+This sets the sensibility when you have to bang your weapon to unjamm it.
+
+```cpp
+#define CLASH_THRESHOLD_G 2.0
+```
+
+### VOLUME
+Then we specify the volumes. Generally values between 0 and 3000 are useful, but it may depend on what kind of board you have.
+
+```cpp
+#define VOLUME 1500
+```
+
+### Standard Features
+Finally, we have a set of defines that enable standard features.
+It's on my TODO list to make these not required:
+
+```cpp
+#define ENABLE_AUDIO
+#define ENABLE_MOTION
+#define ENABLE_WS2811
+#define ENABLE_SD
+```
+
+## Optional Defines
+
+### FILTER_CUTOFF_FREQUENCY
+If present, these defines enable a butterworth highpass filter with the given order and cutoff frequency. The idea is to remove frequencies that your speaker can't reproduce anyways to put less stress on the speaker. The filter order defaults to 8. A reasonable cutoff frequncy might be 100Hz. The filter does not affect I2C or S/PDIF output if enabled.
+
+```cpp
+#define FILTER_CUTOFF_FREQUENCY
+#define FILTER_ORDER
+```
+
+### NO_REPEAT_RANDOM
+This define will make ProffieOS remember which file (of each effect) was played last and avoid playing it again next time. If there are only 2 of a particular sound, there is still a chance that the same sound is played to avoid simply flipping between them in a predictable pattern.
+
+```cpp
+#define NO_REPEAT_RANDOM
+```
+
+### ENABLE_BLASTER_AUTO
+This define determines if your weapon has the automatic fire mode enabled. If you want your weapon to be semi-automatic only, simply comment it out.
+```cpp
+#define ENABLE_BLASTER_AUTO
+```
+
+### BLASTER_SHOTS_UNTIL_EMPTY
+This is your weapon round capacity per "clip" (i.e. before needing to reload). If you want an infinite ammo weapon, simply comment it out.
+```cpp
+#define BLASTER_SHOTS_UNTIL_EMPTY 15
+```
+
+### BLASTER_JAM_PERCENTAGE
+Here we set the change of the weapon jamming per shot. The range is 0-100. Please note that if you comment it out, it will be replaced with a random chance per ignition.
+```cpp
+#define BLASTER_JAM_PERCENTAGE 1
+```
+
+### FEMALE_TALKIE_VOICE
+Uses a female talkie voice, only affects built-in error messages, for other sounds, go check out the Sound Library.
+
+```cpp
+#define FEMALE_TALKIE_VOICE
+```
+
+## Config Sections Samples
```

Comment:
Seems kind of redundant to go through all the sections, and then also have everything together.  I wonder if maybe it would be better to do it the other way around: Start with the config file, and then discuss the various segments. That way there may be less repetition?